internal/stackdiag.go

```diff
@@ -166,12 +160,26 @@ func (ds *diagJobState) scheduleJob(typ, esName, resourceName string, tls bool)
 	if err != nil {
 		return err
 	}
-	ds.jobs[podName] = &diagJob{
-		podName:      podName,
-		namespace:    ds.ns,
-		resourceName: resourceName,
-		typ:          typ,
+
+	var job = diagJob{
+		RemoteSource: extraction.RemoteSource{
+			Namespace:    ds.ns,
+			PodName:      podName,
+			Typ:          typ,
+			ResourceName: resourceName,
+			PodOutputDir: podOutputDir,
+		},
 	}
+	// start a dedicated timer for each job and terminate the job when the timer expires.
+	go func(j *diagJob) {
+		timer := time.NewTimer(jobTimeout)
```

Comment:
I think this timer must be stopped at some point ? Otherwise it will eventually fire if all the other diagnostics are not retrieved within the next 5 minutes ?